File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        for(int i = 0; i < photo.length; i++){
            int score = 0;
            for(int j = 0; j < name.length; j++){
                if(Arrays.asList(photo[i]).contains(name[j])) score += yearning[j];
            }
            answer[i] = score;
        }
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    let answer = [];
    for(let i = 0; i < photo.length; i++){
        let score = 0;
        for(let j = 0; j < name.length; j++){
            if(photo[i].includes(name[j])) score += yearning[j];
        }
        answer.push(score);
    }
    return answer;
}